python/composio/cli/workspaces.py

```diff
@@ -0,0 +1,138 @@
+"""
+Composio workspace manager.
+"""
+
+import typing as t
+from datetime import datetime
+from uuid import uuid4
+
+import click
+
+from composio.cli.context import Context, pass_context
+from composio.cli.utils.decorators import handle_exceptions
+from composio.cli.utils.params import EnumParam
+from composio.client.collections import ComposioWorkspaceStatus
+from composio.client.exceptions import HTTPError
+
+
+@click.group(name="workspaces", invoke_without_command=True)
+@click.option(
+    "--status",
+    type=EnumParam(
+        cls=ComposioWorkspaceStatus,
+    ),
+    help="Workspace status filter.",
+    required=False,
```

Comment:
The method `_get_workspace_id` in `workspaces.py` uses the `id` variable in the error message before it's checked for `None`. This could lead to misleading error messages if `id` is `None`. Consider checking if `id` is `None` before using it in the error message.